src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -85,8 +87,40 @@ public List<Answer> getAnswers() {
         return answers;
     }
 
+    public List<DeleteHistory> deleteBy(NsUser user) throws CannotDeleteException {
+        List<DeleteHistory> deleteHistories = new ArrayList<>();
+        if (!this.isOwner(user)) {
+            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
+        }
+
+        if (this.getAnswers().stream()
+            .anyMatch(a -> !a.isOwner(user))
+        ) {
+            throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
+        }
+
+        this.getAnswers().forEach(a -> {
+            try {
+                deleteHistories.add(
+                    new DeleteHistory(ContentType.ANSWER, a.getId(), user, LocalDateTime.now()));
+                a.deleteBy(user);
```

Comment:
답변 삭제 이력은 Answer 객체가 반환해줄 수 있지 않을까요?
```suggestion
                DeleteHistory answerDeleteHistory = a.deleteBy(user);
                deleteHistories.add(answerDeleteHistory);
```